### PR TITLE
CRUD: Add CRUD operations for ajax functions.

### PR DESCRIPTION
**Pull request by coderkevin**
Update for #12324 

This commit changes post and post_meta calls to CRUD objects
throughout the wc-ajax server-side code.

It does not update the most complex functions that have inline SQL.

### CODE REVIEW

File: includes/class-wc-ajax.php [hunk 1]
```diff
@@ -2240,21 +2227,38 @@ private static function variation_bulk_adjust_price( $variations, $field, $opera
...
 	 * @access private
 	 * @param array $variations
 	 * @param string $field
 	 * @param string $value
 	 */
-	private static function variation_bulk_set_meta( $variations, $field, $value ) {
+	private static function variation_bulk_set( $variations, $field, $value ) {
+		foreach ( $variations as $variation_id ) {
+			$variation = wc_get_product( $variation_id );
+			$variation->{ "set$field" }( wc_clean( $value ) );
```
Comment (mikejolley) [inline includes/class-wc-ajax.php]: It would make more sense to pass the actual prop name than the _ prefixed meta key. You can add that here.

**Review by mikejolley — COMMENTED** [review]
Few questions and suggestions added :) Thanks!

File: includes/class-wc-ajax.php [hunk 2]
```diff
@@ -808,6 +812,7 @@ public static function grant_access_to_download() {
 
 	/**
 	 * Get customer details via ajax.
+	 * @todo CRUD (customer)
```
Comment (coderkevin) [inline includes/class-wc-ajax.php]: Yes, I think SQL should go into the DataStore implementations. Thanks!

File: includes/class-wc-ajax.php [hunk 3]
```diff
@@ -1946,12 +1957,13 @@ public static function save_variations() {
...
-
 		foreach ( $variations as $variation_id ) {
-			$post_status = get_post_status( $variation_id );
-			$new_status  = 'private' === $post_status ? 'publish' : 'private';
-			$wpdb->update( $wpdb->posts, array( 'post_status' => $new_status ), array( 'ID' => $variation_id ) );
+			$variation = wc_get_product( $variation_id );
+			$variation->set_status( 'private' === $variation->get_status( 'edit' ) ? 'publish' : 'private' );
+			$variation->save();
+
+			$variation = wc_get_product( $variation_id );
```
Comment (coderkevin) [inline includes/class-wc-ajax.php]: Oops, that was the remnants of some debugging code. It's removed now. Thanks!

**Pull request merged**